src/test/java/webdrivertasks/test/GoogleCloudTests.java

```diff
@@ -16,26 +15,38 @@
     private final String MACHINE_TYPE = "e2-standard-8";
     private final String MANUALLY_CHECKED_TOTAL_COST = "6,940.38";
 
+
     @BeforeClass(alwaysRun = true)
     public void browserSetup() {
         driver = WebDriverSetUp.getDriver();
         step = new Steps(driver);
     }
 
     @Test
-    public void chekHurtMePlentyTask() {
-
-        step.goToCalculatorPage();
-        step.selectEngineProperties();
-        step.selectNodeProperties();
-        step.calculateTotalCost();
+    public void checkHurtMePlentyTask() {
+        CalculatorPage calculatorPage = step.goToCalculatorPage()
+                .selectEngineProperties()
+                .selectNodeProperties()
+                .estimateTotalCost();
+
+        Assert.assertTrue(calculatorPage.checkTotalEstimatedCost(MANUALLY_CHECKED_TOTAL_COST));
+        Assert.assertTrue(calculatorPage.checkMachineClass(MACHINE_CLASS));
+        Assert.assertTrue(calculatorPage.checkInstanceType(MACHINE_TYPE));
+    }
 
-        Assert.assertTrue(step.calculateTotalCost().checkTotalEstimatedCost(MANUALLY_CHECKED_TOTAL_COST));
-        Assert.assertTrue(step.calculateTotalCost().checkMachineClass(MACHINE_CLASS));
-        Assert.assertTrue(step.calculateTotalCost().checkInstanceType(MACHINE_TYPE));
+    @Test
+    public void checkHardcoreTask() {
+        String totalCostFromEmail = step.startFromCalculatorPage()
+                .selectNodeProperties()
+                .estimateTotalCostToCompareOneFromEmail()
+                .openNewTenMinuteMailPageAndSwitchIt()
+                .sendEmail()
+                .getTotalCostFromEmail();
+
+        Assert.assertTrue(new CalculatorPage(driver).switchToInternalFrame().checkTotalEstimatedCost(totalCostFromEmail));
```

Comment:
this logic looks quite complicated, don't you think? I'd suggest moving it to a separate function